code/datums/gas_giant.dm

```diff
@@ -0,0 +1,14 @@
+var/global/datum/gas_giant/gas_giant
+
+/datum/gas_giant
+	var/name
+	var/datum/gas_mixture/GM
+
+/datum/gas_giant/New()
+	name = take_name(planet_names)
+	GM = new
+	GM.temperature = rand(T0C-224, T0C+95)
+	GM.adjust_multi(
+		GAS_OXYGEN, rand(10, 50)*1000/(R_IDEAL_GAS_EQUATION * GM.temperature),
+		GAS_NITROGEN, rand(10, 50)*1000/(R_IDEAL_GAS_EQUATION * GM.temperature),
+		GAS_PLASMA, rand(5, 25)*1000/(R_IDEAL_GAS_EQUATION * GM.temperature))
```

Comment:
Adding limits to gas giant gas supplies seems like a bad idea to me.  We have infinite gas supplies for a reason.

Reply:
How else should I do this?

Reply:
While a huge atmos issue would likely just get a shuttle called if they were to declare WE BUILD I don't think finite gases are a good idea. A ton of air gets wasted by vents just pumping into breached areas and theres no way to stop this so all of the air you worked hard to get is now up in smoke.
Filling rooms is already a huge pain in the ass with most people opting to call it rather than fix it more often than not. With this atmos can be forced into the situation of "lol we don't even have gas go fill it so Im going to go toss myself into the hacked SSU" which sounds like something Id rather not see.

Reply:
It's unlimited, don't worry